siaskynet/_upload.py

```diff
@@ -48,41 +109,12 @@ def upload_file_request(self, path, custom_opts=None):
         return None
 
     with open(path, 'rb') as file_h:
-        filename = opts['custom_filename'] if opts['custom_filename'] \
-            else os.path.basename(file_h.name)
-        files = {opts['portal_file_fieldname']: (filename, file_h)}
+        if not opts['custom_filename']:
+            opts['custom_filename'] = os.path.basename(file_h.name)
```

Comment:
I would use a temporary variable here instead of overriding the values in `opts`. See the Go SDK

Reply:
Oh, I see why you do this. The `filename` parameter is only needed for uploading directories, so this comment is correct:

```py
        # this appears to be missing in go sdk; maybe the server ignores it?
```

Reply:
And if someone passes a map of filenames to files (e.g. `upload_data`) then we can assume the filename there is the one he wants to use. `custom_filename` should only apply to `upload_file` and not to `upload`, IMO.

Reply:
Note that `opts` is a copy, so overriding is invisible to the user.  I'll use a temporary variable like in the go sdk, though.